### DISCUSSION TITLE
Application error: a client-side exception has occurred .

### DISCUSSION THREAD

**Discussion by tsimvris** [neuefische/questions#127] [Web] [answered]
Hi,

ich habe eine neue Fehlermeldung , yuhu 🥇 .

Das problem taucht nur bei der Vercel deployment auf , sobald man die charts mit Daten "befühlt" und auf der home page geht kriegt mal der Meldung das Bar keine registered element ist, obwohl es ist.

Fehlermeldung: 
<img width="1157" alt="Bildschirmfoto 2022-08-30 um 14 00 17" src="https://user-images.githubusercontent.com/101249132/187439340-92c4953a-80da-4aef-8a1c-1ead7aba2569.png">

Code :
<img width="535" alt="Bildschirmfoto 2022-08-30 um 14 43 57" src="https://user-images.githubusercontent.com/101249132/187439719-72d50057-468d-4e37-8399-144d65e2ee52.png">

Hat jemand Erfahrung mit ChartJS and hat diese Fehler bereits gesehen ?

**Comment by mpagels** [comment]
Hey @tsimvris, magst du dein repo hier noch verlinken?

**Reply by tsimvris** [reply]
Sorry,

[hier]( https://github.com/tsimvris/capstone-project/tree/homepage) ist es

**Reply by mpagels** [reply]
Das Vercel Deployment will, dass ich mich einlogge? Hast du ein User, den ich nutzen kann, damit ich den Anmelde Prozess nicht machen muss?

**Reply by tsimvris** [reply]
klar, testUser , pw= testUser

**Reply by tsimvris** [reply]
um den fehler zu reproduzieren musst du

**Reply by tsimvris** [reply]
3 clients hinterlegen, auf profile die infos füllen, eine service hinterlegen, und min 3 invoices schreiben, und dann auf home gehen

**Reply by tsimvris** [reply]
Sorry :)

**Comment by j-eick** [comment]
I'm also using Chart.js and I've got the exact same problem with my Line/Bar-Chart. Both lines and bars are in one chart and Vercel throws me the same error. `Error: "bar" is not a registered controller.`

**Comment by alexclaes** [comment, marked as answer]
@tsimvris I can reproduce this error with your code.

The error says `Error: "bar" is not a registered controller.` I'm not sure why this only happens with the production build and not with dev mode. It seems the solution is to simply register the bar controller.

```js
import {
	Chart as ChartJS,
	BarController,
	/* ... other ... */
} from 'chart.js';

ChartJS.register(
	BarController,
	/* ... other ... */
);
```

You can test the production build on your machine, so you don't need to wait for the PR checking und Vercel deployment to see if it is working.

```
npm run build
```
Wait for the  production build to finish, then start it:

```
npx next start
```

You can open http://localhost:300 in your browser as with the dev mode, but now it's the same code that would run on vercel. 

Don't forget to create a new build und start it after every(!) change in the code, if you like to test the production build.

**Reply by tsimvris** [reply]
Thank you very much,

The Barcontroller import and register is working.

👍🏼 👍🏼 👍🏼 👍🏼 👍🏼 👍🏼
